.github/workflows/default.yml

```
@@ -11,26 +11,41 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
+      # Check-out repo and set-up python
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Set up Python 3.12.5
+      - name: Set up Python 3.12
+        id: setup-python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.12.5
-      - name: Install Python dependencies
-        run: |
-          python3 -m venv venv
-          source ./venv/bin/activate
-          pip install -U pip
-          pip install -r ./requirements.txt
+          python-version: 3.12
+      # Install & configure poetry
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          virtualenvs-path: .venv
+          installer-parallel: true
+      # Load cached venv if cache exists
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v4
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+      # Install dependencies if cache does not exist
+      - name: Install dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-root
```

If we go with 
```
[tool.poetry]
package-mode = false
```
We can skip this - https://python-poetry.org/docs/basic-usage#operating-modes
```suggestion
        run: poetry install --no-interaction
```

---

yeah, I know that, it's in my todo list, thx